packages/flutter/lib/src/foundation/shader_warm_up.dart

```diff
@@ -0,0 +1,86 @@
+import 'dart:ui';
+
+import 'package:flutter/src/material/colors.dart';
+
+/// Signature to warm up GPU shader compilation cache.
+///
+/// To warm up shaders for a specific draw function, call it with the provided
+/// [canvas]. For example, to warm up path rendering shaders, call
+/// `canvas.drawPath` with desired paths. Assume that [canvas] has a size of
+/// 1000x1000. Drawing outside the range may be clipped.
+///
+/// See also [defaultShaderWarmUp].
+typedef ShaderWarmUp = void Function(Canvas canvas);
+
+/// Trigger common draw operations to warm up GPU shader compilation cache.
+///
+/// When Skia first sees a certain type of draw operations on GPU, it needs to
+/// compile the corresponding shader. The compilation can be slow (20ms-200ms).
+/// Having that time as a startup latency is much better than having a jank in
+/// the middle of an animation.
+///
+/// Therefore we use this (by default) in [SchedulerBinding.scheduleWarmUpFrame]
+/// to move common shader compilations from animation time to startup time.
+/// Alternatively, [customShaderWarmUp] can be provided to [runApp] to replace
+/// this default warm up function.
+ShaderWarmUp defaultShaderWarmUp = (Canvas canvas) {
```

Comment:
You can define the function as `void defaultShaderWarmUp(Canvas canvas) {` and it will still be assignable to `ShaderWarmUp`

Reply:
I intentionally wrote `ShaderWarmUp` here to clarify that the canvas is expected to have a size of 1000x1000. (Otherwise, I could delete `ShaderWarmUp` and just use the signature directly everywhere.)

Reply:
I mean in terms of how the function is declared:

```suggestion
void defaultShaderWarmUp(Canvas canvas) {
```

Reply:
I understood that. My intention is that when developers see the signature of `defaultShaderWarmUp`, they can discover a link to `ShaderWarmUp` and realize that we're doing work on a 1000x1000 canvas. Does that help?

Reply:
If you leave a comment in the dartdoc `[ShaderWarmup]` they can follow that link as welll

Reply:
Done. Thanks!